### PR TITLE
Fix IMDaemonCore nil pointer on iOS

### PR DESCRIPTION
**Pull request by CarlAmbroselli**
Fixes the crash from https://github.com/open-imcore/barcelona/issues/35.

This fixes receiving i.e. link attachments, but still fails (but not crashes) for images. Will still look into that, but fixing the crash was most critical for now.

### CODE REVIEW

File: Core/Barcelona/Queries/IMDPersistence.swift [hunk 1]
```diff
@@ -228,7 +228,13 @@ public func BLLoadChatItems(_ items: [(chatID: String, messageID: String)]) -> P
 
 typealias IMFileTransferFromIMDAttachmentRecordRefType = @convention(c) (_ record: Any) -> IMFileTransfer?
 
-private let IMDaemonCore = "/System/Library/PrivateFrameworks/IMDaemonCore.framework/Versions/Current/IMDaemonCore".withCString({
+#if os(iOS)
+let IMDaemonCoreLocation = "/System/Library/PrivateFrameworks/IMDaemonCore.framework/IMDaemonCore"
```
Comment (KishanBagaria) [inline Core/Barcelona/Queries/IMDPersistence.swift:232]: seems like the conditional isn't necessary, this location works well on macOS too

Comment (CarlAmbroselli) [reply Core/Barcelona/Queries/IMDPersistence.swift:232]: Ah cool, I don't have it setup on Mac, so unable to test it

Comment (KishanBagaria) [reply Core/Barcelona/Queries/IMDPersistence.swift:232]: Screen Shot 2022-02-17 at 2 33 03 PM

on macOS 12.2.1

Comment (CarlAmbroselli) [reply Core/Barcelona/Queries/IMDPersistence.swift]: PR updated

**Pull request merged**